src/events/marketEvents.js

```diff
@@ -7,69 +7,79 @@ const { getTokenName, getTokenSymbol } = require('../services/contractERC20');
-                            .then((questions) => {
-                                if (questions.length == 0) {
-                                    console.error(`ERROR: Question for hex "${condition.questionId}" not found on contition ID ${event.returnValues.conditionIds[0]}`);
-                                } else {
-                                    message.push(questions.map(question => 
-                                        `> *<https://omen.eth.link/#/${condition.fixedProductMarketMakers}|${question.title}>*\n> *Outcomes:*`
-                                    ));
-                                    questions.forEach(question => {
-                                        if (condition.outcomeTokenMarginalPrices) {
-                                            message.push(
-                                                question.outcomes.map((outcome, i) => 
-                                                    `> \`${(parseFloat(condition.outcomeTokenMarginalPrices[i]) * 100 )
-                                                        .toFixed(2)}%\` - ${outcome}`
-                                                ));
-                                        } else {
-                                            message.push(
-                                                question.outcomes.map(outcome => `> - ${outcome}`));
-                                        }
-                                    });
-                                }
-                                message.push(`> *Collateral*: <https://${urlExplorer}/token/${event.returnValues.collateralToken}|${tokenName}>`,
-                                    `> *Liquidity*: ${parseFloat(condition.scaledLiquidityParameter).toFixed(2)} ${tokenSymbol}`,
-                                    `> *Created by*: <https://${urlExplorer}/address/${transaction.from}|${truncate(transaction.from, 14)}>`);
-                                    // Send Slack notification
-                                pushSlackArrayMessages(message);
-                                console.log(event.returnValues.conditionIds[0] + ':\n' + message.join('\n') + '\n\n');
-                            });
+        fromBlock,
+        toBlock,
+    }, (error, events) => {
+        events.forEach(event => {
+            (async () => {
+                const message = new Array('<!here>', '*New market created!* :tada:');
+                //TODO support conditional markets
+                Promise.all([
+                    getCondition(event.returnValues.conditionIds[0]),
+                    web3.eth.getTransaction(event.transactionHash),
+                    getTokenName(web3, event.returnValues.collateralToken), 
+                    getTokenSymbol(web3, event.returnValues.collateralToken), 
+                ])
+                    .then(([conditions, transaction, tokenName, tokenSymbol]) => {
+                        conditions.forEach(condition => {
```

Comment:
If the purpose of this , is to generate an array of messages, you can iterate over a map with async / await, so you avoid to use a lot of `then`

Reply:
Thans Mariano! I will add these suggestion in other new PR

Reply:
https://github.com/gnosis/conditional-tokens-bot/issues/30